### DISCUSSION TITLE
Background in Perl programming

### DISCUSSION THREAD

**Discussion by eltikia** [eltikia/Perl-1.0#56] [Q&A] [answered]
What's your background in Perl programming?

**Comment by Stagyrite** [comment, marked as answer]
I had a Perl course when I studied Computer Science at the [University of Wrocław](https://www.ii.uni.wroc.pl/ "Instytut Informatyki Uniwersytetu Wrocławskiego"). Later I used multiple scripting languages during my studies to solve mathematical assignments. When at work, I tried modern languages like [Ruby](https://www.ruby-lang.org/en/ "Ruby Programming Language"), [Python](https://www.python.org/ "Welcome to Python.org"), and [Scala](https://www.scala-lang.org/ "The Scala Programming Language"). Those are similar to Perl, although I still use Java for practical text extraction. My Perl Geek Code is turned into tooltiped links with a Java project, although the code was based on [a Perl converter](https://www.xmltwig.org/pgc/ "The Perl Geek Code").

-----BEGIN PERL GEEK CODE BLOCK-----  
Version: 0.01  
[P++>+++](#perl-geek-code "Perl Programmer Stage: Adept. But someday I would like to say: Hacker.") [c---](#perl-geek-code "Core Hacker: I sure don't want to mess with Perl development") [P6](#perl-geek-code "Perl 6: I subscribed to at least one of the Perl 6 mailing lists") [R++](#perl-geek-code "Regular Expressions: I use ///e and I know the difference between //m and //s") [M](#perl-geek-code "Modules: I use modules from CPAN") [O](#perl-geek-code "Objects: I have used other people's OO modules") [MA+](#perl-geek-code "Map: I often try to use map, but as soon as I have a problem in the code I replace it with a foreach") [E-](#perl-geek-code "Eval: I know the difference between eval &quot;string&quot; and eval { block }") [PU-](#perl-geek-code "Pack/Unpack: I try to avoid pack and unpack, I don't feel real confident with them") [BD](#perl-geek-code "Bondage & Discipline: I use -w and strict for anything but one-liners") [C-](#perl-geek-code "CGI: CGI is best done with print and parsing CGI queries by hand") [D!](#perl-geek-code "DBI: I don't need no stinkin' data base!") [S-](#perl-geek-code "Sysadmin: I know where the logs are, why do i need perl?") [X-](#perl-geek-code "XML: Yeah I know, XML is great, but why is it such a pain to use?") [WP-](#perl-geek-code "Perlmonk: I am not interested in Perl Monks") [MO!](#perl-geek-code "Perl Monger: There is no Perl Mongers group in my area") [PP](#perl-geek-code "#perl: I occasionally visit #perl") [n!](#perl-geek-code "Usenet: I have given up on usenet in general and the c.l.p.m groups, they're useless")
 [CO!](#perl-geek-code "Conferences: I have never attended TPC or YAPC, and I have no desire to ever do so") [PO-](#perl-geek-code "Perl Poetry: I don't write Perl Poetry but I like reading it or I use Coy.pm occasionnally") [o!](#perl-geek-code "JAPH/Obfuscation: I HATE obfuscation") [G](#perl-geek-code "Goof: Perl is fun, I try not use the most exotic features though, someone will have to maintain my code some day") [A-](#perl-geek-code "Abuse/Glue: Writing portable code is hard enough") [OLC++](#perl-geek-code "C: I use C quite often and I like it as much as I like Perl") [OLCC-](#perl-geek-code "C++: I have to use C++, it does not mean that I like it") [OLJ*](#perl-geek-code "Java: I am working in a basement at Sun") [OLP-](#perl-geek-code "Python: I have to use Python, it does not mean that I like it") [OLR-](#perl-geek-code "Ruby: I have to use Ruby, it does not mean that I like it") [OLCO-](#perl-geek-code "COBOL: I have to use COBOL, it does not mean that I like it") [OLS-](#perl-geek-code "Smalltalk: I have to use Smalltalk, it does not mean that I like it") [OLL-](#perl-geek-code "Lisp: I have to use Lisp, it does not mean that I like it")
 [OLA-](#perl-geek-code "Assembly: I have to use Assembly, it does not mean that I like it") [Ee-](#perl-geek-code "Emacs: Emacs is too big and bloated for my tastes") [Ev+](#perl-geek-code "Vi: I can open multiple files and use shell escapes effectively") [Eon](#perl-geek-code "Notepad: I use Notepad when I really have to") [Eot!](#perl-geek-code "Textedit: I refuse to use Textedit") [Eob!](#perl-geek-code "BBEdit: I refuse to use BBEdit") [Eoa!](#perl-geek-code "Alpha: I refuse to use Alpha") [uL+](#perl-geek-code "Linux: I use Linux amongst others") [uB!](#perl-geek-code "BSD: Why would I care about BSD?") [uS!](#perl-geek-code "Sun OS/Solaris: Why would I care about Sun OS/Solaris?") [uH!](#perl-geek-code "HPUX: Why would I care about HPUX?") [uo!](#perl-geek-code "An Other Unix: Why would I care about An Other Unix?") [w+](#perl-geek-code "Windows: I mostly use Windows, I like it") [m!](#perl-geek-code "MacIntosh: Gimme back my command line!") [osA](#perl-geek-code "Amiga: I occasionnaly use Amiga") [osBE-](#perl-geek-code "BeOS: I try to avoid using BeOS, frankly it sucks!")
------END PERL GEEK CODE BLOCK------
